### ISSUE TITLE
Release 4.6.0 - No temps for Powerwall 3

### ISSUE THREAD

**Issue by jasonacox** [jasonacox/Powerwall-Dashboard#550]
I upgraded to the 4.6.0 release and unfortunately I am not getting any temperature readings from my PW3. Dashboard says "No data" and http://pypowerwall:8675/temps returns "{}". Seems like maybe something is different with PW3. Is there anything I can do to help debug?

_Originally posted by @jplewis2 in https://github.com/jasonacox/Powerwall-Dashboard/discussions/392#discussioncomment-11432352_

**Comment by jasonacox** [comment]
Thanks @jplewis2 take a look at http://pypowerwall:8675/tedapi/controller and scroll down. Do you see anything that looks like this? If you can, share the payload you get (remove serial numbers).

```json
{
"activeAlerts": [],
"partNumber": "2012170-25-E",
"serialNumber": "TG1213456778901",
"signals": [
{
"boolValue": null,
"name": "THC_AmbientTemp",
"textValue": null,
"timestamp": "2024-12-01T21:37:45-08:00",
"value": 18.900000000000006
}
]
},
```

**Comment by farberm** [comment]
I have the same output for temp. 

PW 3, Firmware 24.36.1. Gateway info etc...

[proxy] [INFO] pyPowerwall [0.12.0] Proxy Server [t66] - HTTP Port 8675
proxy] [INFO] pyPowerwall Proxy Started
[proxy] [INFO] pyPowerwall Proxy Server - Local Mode
 [proxy] [INFO] Connected to Energy Gateway 192.168.91.1 (Home)
[proxy] [INFO] TEDAPI Mode Enabled for Device Vitals (full)

This is a direct connection thru a LAN port. Redirect thru unifi static route. 
          {
            "boolValue": null,
            "name": "THC_AmbientTemp",
            "textValue": null,
            "timestamp": null,
            "value": null
          }

**Comment by jplewis2** [comment]
I am getting the same as @martinmanyhats and @farberm for my output:

```json
      {
        "activeAlerts": [],
        "partNumber": "",
        "serialNumber": "",
        "signals": [
          {
            "boolValue": null,
            "name": "THC_AmbientTemp",
            "textValue": null,
            "timestamp": null,
            "value": null
          }
        ]
      },
```

**Comment by jasonacox** [comment]
Thank you all. 

It seems like this is a consistent report from Powerwall 3 owners. I haven't heard of a working PW3 yet.

There is a possibility that the temperature sensor was removed from the PW3, but that seems unlikely as they would want it on the CAN bus to trigger resistance heat during freezing weather conditions (required to make the batteries work). It could be stored under a different payload which also seems odd if they are both using the Tesla One app for setup and troubleshooting.

Are  any of you with Powerwall 3 systems able to see temperature data in the Tesla One app?  If so, if you are able to analyze the network traffic during that session, it may point us toward the right query needed to get PW3 temps.

I should update the PR and notes to indicate that the 4.6.0 release only works for PW, PW2 and PW+ systems.

**Comment by petersloan** [comment]
I can confirm that in Tesla One with Powerwall 3 and Gateway 2, there are no temps anywhere that I can find.  If someone can confirm where exactly in Tesla One you find temps, I will check again.

I would have expected them in System -> Powerwall where the vitals are listed.  Its possible that they are no longer exposed with PW3.

**Comment by jplewis2** [comment]
Yeah, I also can't find temp information anywhere (PW3 + GW2). According to @ygelfand who provided the code to get temp information, he actually looked at the source code for Tesla One to find the new protobuf fields that provide the temperature  - he's never used the app himself: https://github.com/jasonacox/Powerwall-Dashboard/discussions/392#discussioncomment-11477877

It's possible the app is querying the data anyway, just not displaying it. So maybe a network traffic capture would still work? But that seems like a lot of work for something that seems unlikely.

**Comment by ygelfand** [comment]
just to clarify, the app doesn't actually use/collect it, it just opens up the avenue for it.  I don't have a pw3 to test with unfortunately, only have2+'s.    I know  powerwall3 is combined pod/pinv? does it have a separate device type?

**Comment by jplewis2** [comment]
@ygelfand You said you looked at the source code for Tesla One to find the temp data - does that mean you used a tool to disassemble/decompilation the app binaries?

**Comment by ygelfand** [comment]
I did the android app, its mostly react-native, so has alot of js assets.

**Comment by cstout1** [comment]
I'm unsure if this is directly related or needs to be its own issue, but I just upgraded one location's dashboard to latest (Powerwall-Dashboard 4.6.5 on MacOS) from v4.6.3 and Powerwall Temps is now blank. This location does not have any Powerwall 3s, only Powerwall 2s.

http://127.0.0.1:8675/temps shows
`{}`

What's odd to me is that it only happened at one location. The other install I have powerwall-dashboard configured (near identically) did not lose powerwall temps.

**Comment by jasonacox** [comment]
Can you run `verify.sh`?  Temps do require you to be using TEDAPI in hybrid or full mode.

**Comment by cstout1** [comment]
`-e Verify Powerwall-Dashboard 4.6.5 on MacOS - Timezone: America/Los_Angeles
-e Checking pypowerwall
-e ----------------------------------------------------------------------------
-e -n  - Config File pypowerwall.env: 
-e GOOD
-e -n  - Container (pypowerwall): 
-e GOOD
-e -n  - Service (port 8675): 
-e GOOD
-e  - Version: 0.12.7 Proxy t68
-e  - Powerwall State: CONNECTED - Firmware: 25.2.0
-e  - Site Name: Home Energy Gateway
-e  - Powerwall Gateway TEDAPI: Not Available (192.168.91.1)
-e  - Cloud Mode: NO
-e 
...[removed/shortened output to TEDAPI section]...
-e All tests succeeded.`

So, it appears to have reverted to the default API IP address. This gateway is connected to the LAN and reachable by the host. What's odd is the power flow meter is loading fine in the dashboard...

**Comment by cstout1** [comment]
The successful local API connection did not persist after a reboot today. It switched over to Solar-only mode and verify.sh now reports:

> -e ----------------------------------------------------------------------------
> -e -n  - Config File pypowerwall.env: 
> -e GOOD
> -e -n  - Container (pypowerwall): 
> -e GOOD
> -e -n  - Service (port 8675): 
> -e GOOD
> -e  - Version: 0.12.7 Proxy t68
> -e  - Powerwall State: CONNECTED - Firmware: SolarOnly
> -e  - Site Name: {"pypowerwall": "0.12.7 Proxy t68", "mode": "Local", "gets": 714, "posts": 0, "errors": 0, "timeout": 0, "uri": {"/temps/pw": 101, "/strings": 101, "/soe": 101, "/aggregates": 101, "/freq": 101, "/pod": 101, "/alerts/pw": 101, "/version": 1, "/stats": 2}, "ts": 1741740885, "start": 1741740377, "clear": 1741740377, "uptime": "0:08:28", "mem": 36944, "site_name": null, "cloudmode": false, "fleetapi": false, "tedapi": false, "pw3": false, "tedapi_mode": "off", "siteid": null, "counter": 0, "cf": ".auth/.powerwall", "config": {"PW_BIND_ADDRESS": "", "PW_PASSWORD": "**********", "PW_EMAIL": "realEmailRemoved@gmail.com", "PW_HOST": "192.168.1.126", "PW_TIMEZONE": "America/Los_Angeles", "PW_DEBUG": false, "PW_CACHE_EXPIRE": 5, "PW_BROWSER_CACHE": 0, "PW_TIMEOUT": 5, "PW_POOL_MAXSIZE": 15, "PW_HTTPS": "no", "PW_PORT": 8675, "PW_STYLE": "grafana-dark.js", "PW_SITEID": null, "PW_AUTH_PATH": ".auth", "PW_AUTH_MODE": "cookie", "PW_CACHE_FILE": ".auth/.powerwall", "PW_CONTROL_SECRET": null, "PW_GW_PWD": "**********", "PW_NEG_SOLAR": true}}
> -e  - Powerwall Gateway TEDAPI: Not Available (192.168.91.1)
> -e  - Cloud Mode: NO
> -e

**Comment by jasonacox** [comment]
I'm hearing that some systems are upgrading and no longer support the login portal. If you go to 192.168.1.126 in your browser, are you still able to log in?

You will need to switch to the TEDAPI method (see https://github.com/jasonacox/Powerwall-Dashboard?tab=readme-ov-file#extended-device-vitals-metrics) which will have the data you are needing. To confirm access, ping 192.168.91.1. If you are not able to get the route to work (see `add_route.sh` script) you can switch to cloud mode but it has limited data (no temps for one thing).

To switch modes you re-run `setup.sh`.

**Comment by lemassykoi** [comment]
Did you find a solution, or should I remove the empty temps and fans section from dashboard?

PW3 firmware 25.34.3, option 4
powerwall_dashboard version="4.8.5",file_ts="2025-11-17"

let me know if you need me to run some commands

**Comment by lemassykoi** [comment]
Thanks. "Null" everywhere...

<img width="387" height="593" alt="Image" src="https://github.com/user-attachments/assets/dca336fa-5580-41b1-a44b-36fd44747044" />

**Comment by jasonacox** [comment]
@jasonacox-sam I agree with @holstein13  - Please record learning from this issue and store in your KB, then close this issue.

**Comment by jasonacox-sam** [comment]
@jasonacox — done. KB updated with the PW3 temp/fan telemetry finding:

- PW3 does not expose temperature or fan data via TEDAPI at the hardware level — consistent across all firmware versions through 26.18.1
- `/temps` returns `{}`, `/vitals` has no temp/fan fields on any PW3 device
- This is a PW3 hardware limitation, not a dashboard bug
- PW2 users still have full temp/fan telemetry

Thanks @holstein13 for the thorough confirmation across firmware versions, and @lemassykoi for the earlier field verification. Closing this out.

— Sam 🌊
